lib/Reader/readerir.cpp

```diff
@@ -1879,11 +1902,13 @@ bool GenIR::isUnmanagedPointerType(llvm::Type *Type) {
 //
 //===----------------------------------------------------------------------===//
 
-EHRegion *fgNodeGetRegion(FlowGraphNode *Node) { return nullptr; }
-
-EHRegion *fgNodeGetRegion(llvm::Function *Function) { return nullptr; }
+EHRegion *GenIR::fgNodeGetRegion(FlowGraphNode *Node) {
+  return FlowGraphInfoMap[Node].Region;
+}
 
-void fgNodeSetRegion(FlowGraphNode *Node, EHRegion *Region) { return; }
+void GenIR::fgNodeSetRegion(FlowGraphNode *Node, EHRegion *Region) {
+  FlowGraphInfoMap[Node].Region = Region;
```

Comment:
If we never change our minds about block->region association then seems like we could assert that fgNodeGetRegion is nullptr here.

Not sure if that always holds though, given we may need to do things like wrap a try/finally around a synchronized method body. But maybe we could have an Update variant for that.